hashring/rbtree.go

```diff
@@ -259,11 +247,42 @@ func (n *RBNode) search(val int) (string, bool) {
 	return "", false
 }
 
-// Search searches for a value in the RBTree, returns the string and true if found,
-// or the empty string and false if val is not in the tree
-func (t *RBTree) Search(val int) (string, bool) {
+// Search searches for a value in the RedBlackTree, returns the string and true
+// if found or the empty string and false if val is not in the tree.
+func (t *RedBlackTree) Search(val int) (string, bool) {
 	if t.root == nil {
 		return "", false
 	}
 	return t.root.search(val)
 }
+
+// LookupNAt iterates through the tree from the node with value val, and
+// returns the next n unique strings. This function is not guaranteed to
+// return n strings.
+func (t *RedBlackTree) LookupNAt(n int, val int, unique map[string]bool) {
+	dfs(t.root, n, val, unique)
+}
+
+// dfs is a depth-first-search that finds n unique strings
+// with a value bigger or equal than val
+func dfs(node *Node, n int, val int, unique map[string]bool) {
+	if len(unique) >= n || node == nil {
```

Comment:
I think it looks a bit funny to the reader to have two recursive calls both operating on node.right.  Is it cleaner if we do something more like the following?

```
 func dfs(node *Node, n int, val int, unique map[string]bool) {
 	if len(unique) >= n || node == nil {
 		return
 	}
 
 	// don't ivestigate left branch when all values there are smaller than the
 	// target value
 	if node.val > val {
	 	dfs(node.left, n, val, unique)
 	}

	if len(unique) >= n {
		return
	}

 	unique[node.str] = true
        dfs(node.right, n, val, unique) 
 }
```